make: *** [sph2pipe_v2.5/sph2pipe] Error

/Library/Developer/CommandLineTools/usr/bin/make -C sph2pipe_v2.5
yes
checking for gcc option to accept ANSI C... cc -o sph2pipe -s -w -g -O2  file_headers.c shorten_x.c sph2pipe.c -lm
file_headers.c:100:17: error: call to undeclared library function 'strdup' with type 'char *(const char *)'; ISO C99 and later do not support implicit function declarations [-Wimplicit-function-declaration]
            inporder = strdup( fldsval );
                       ^
file_headers.c:100:17: note: include the header <string.h> or explicitly provide a declaration for 'strdup'
1 error generated.
none needed
checking for main in -lm... yes
checking how to run the C preprocessor... sph2pipe.c:213:16: error: call to undeclared library function 'strdup' with type 'char *(const char *)'; ISO C99 and later do not support implicit function declarations [-Wimplicit-function-declaration]
            hdrfile = strdup( optarg );
                      ^
sph2pipe.c:213:16: note: include the header <string.h> or explicitly provide a declaration for 'strdup'
1 error generated.
make[1]: *** [sph2pipe] Error 1
make: *** [sph2pipe_v2.5/sph2pipe] Error 2
make: *** Waiting for unfinished jobs....

--- a/tools/Makefile
+++ b/tools/Makefile
@@ -165,6 +165,11 @@ sph2pipe_v$(SPH2PIPE_VERSION)/Makefile: sph2pipe-$(SPH2PIPE_VERSION).tar.gz
 	rm -rf sph2pipe_v*
 	tar -xmzf sph2pipe-$(SPH2PIPE_VERSION).tar.gz
 	mv sph2pipe-$(SPH2PIPE_VERSION) sph2pipe_v$(SPH2PIPE_VERSION)
+	# Workaround for macOS bug <rdar://problem/19363342>
+	if [ `uname` = "Darwin" ]; then \
+		sed -i -e "s/#define _XOPEN_SOURCE 500/#define _XOPEN_SOURCE 600/g" sph2pipe_v$(SPH2PIPE_VERSION)/sph2pipe.c ; \
+		sed -i -e "s/#define _XOPEN_SOURCE 500/#define _XOPEN_SOURCE 600/g" sph2pipe_v$(SPH2PIPE_VERSION)/file_headers.c ; \
+	fi
 
 sph2pipe-$(SPH2PIPE_VERSION).tar.gz:
 	if [ -d "$(DOWNLOAD_DIR)" ]; then \

